pallets/accounts/src/lib.rs

```diff
@@ -404,16 +429,38 @@ pub mod pallet {
 
 			<EVMLinks<T>>::insert(sender.clone(), eth_key);
 			<EVMLinksReverse<T>>::insert(eth_key, sender.clone());
-			let zero: T::Balance = 0u32.saturated_into();
-			<FragUsage<T>>::insert(sender.clone(), zero);
+
+			// Check if the Ethereum account has already some tickets and nova registered when it was not linked
+			if <EthReservedTickets<T>>::contains_key(&eth_key) {
+				// mint tickets
+				let tickets_amount = <EthReservedTickets<T>>::get(&eth_key).unwrap();
```

Comment:
unwrap? you already wrote state above...
Verify first write after

Reply:
Also no unwraps 🙏

Reply:
Right. I am moving these lines after the if-statements:

```rust
<EVMLinks<T>>::insert(sender.clone(), eth_key);
<EVMLinksReverse<T>>::insert(eth_key, sender.clone());
```